simplemenu/output/Bittboy/deploy.sh

```diff
@@ -2,43 +2,44 @@
 set -e
 echo "Welcome to Taichi's simplemenu deploy script! Initiating..."
 
-# Preparation: clean 'main' partition - delete all macos dot files 
-echo "Deleting all macos dot files"
-find /mnt/ -name '._*' -exec rm -v {} &> /dev/null \;
-echo "Preparation completed"
-
-# Step 0: Copy main.sh to a variable for later use
-if [ -f "/mnt/apps/simplemenu/main.sh" ]; then
-    echo "Saving main.sh contents from /mnt/apps/simplemenu/"
-    cat_main_content=$(cat /mnt/apps/simplemenu/main.sh)
-    echo "Step 0 completed"
+# Preparation for MacOS users: Find and remove dot files generated by MacOS. 
+if find /mnt/ -name '._*' -print -quit | grep '._' >/dev/null; then
```

Comment:
I'm wondering what you are trying to achieve here by using the same `find` twice. Basically if there are no ._ files, both `find` commands (line 6 and 8) have the same result (because the `exec` part is only ran for files which are found). So it equals to running the comment without any condition (as in the previous code).
The difference in your case is that if there are ._ files present, you run the search up until the first ._ file again on line 8 because the search starts from scratch.  
So adding the condition like this here has no useful purpose...
It probably doesn't matter in this application, but good to think about optimization...
¯\_(ツ)_/¯

Reply:
'it stops and proceeds to start removing' that is inaccurate. It stops and proceeds to start a new search by `find` followed by removing. But `find` starts searching from beggining (effectivelly searching again the same portion until the first file is found again). It is not using any information from the previous `find` search on line 6. So no, this will not cause it to be quicker.

`it's either this or not doing anything at all.` You don't understand my point. You can completely omit the condition test, as it doesn't bring anything in comparison with not having it, beside an extra line of code ;)

Reply:
It's up to you what you do with the information. I only point it out because I think it is good to know how the script flows as it may prove signifficant in other projects...